src/project.ts

```diff
@@ -152,3 +172,28 @@ export function path(
   }
   return "/" + result.join("/");
 }
+
+// We need a sha1 commit for the API call so we return early if rev is a sha1 and else we look into branches.
+// As fallback we go for the head
+// TODO: Add tags parsing
+export function parseRevision(head: string, revision: string, branches?: [string, string][]): string {
+  if (isOid(revision)) return revision;
+  if (branches) {
+    const branch = branches.find(([name,]) => name === revision);
+    return branch ? branch[1] : head;
+  }
+  return head;
+}
+
+// Splits the path consisting out of an eventual sha1 commit or branch name and the path into a tuple [revision, path]
+export function splitPrefixFromPath(input: string, branches: [string, string][], head: string): [string, string] {
+  const branch = branches.find(([branchName,]) => input.includes(branchName));
```

Comment:
We can be more specific here, because the branch should always be the prefix, so I think `input.indexOf(branchName) === 0` is better. It may also prevent some errors.

Reply:
Yeah, I think eventually to be more explicit we could use `String.startsWith`